File: 프로그래머스/0/120866. 안전지대/안전지대.js
function solution(board) {    
    for(let i=0; i<board.length; i++) {
        for(let j=0; j<board[i].length; j++) {
            if(board[i][j]===1) {
                for(let k=-1; k<2; k++) {
                    for(let l=-1; l<2; l++) {
                        let x = i+k;
                        let y = j+l;
                        if(x>=0 && x<board.length && y>=0 && y<board[i].length && board[x][y]===0) {
                            board[x][y] = 2;
                        }
                    }
                }
            }
        }
    }
    
    return board.reduce((a, b) => [...a,...b]).filter((x) => x===0).length;
}

File: 프로그래머스/0/120866. 안전지대/안전지대.py
def solution(board):
    for i, line in enumerate(board):
        for j, cell in enumerate(line):
            if cell == 1:
                for danger_i in range(i-1, i+2):
                    for danger_j in range(j-1, j+2):
                        if 0 <= danger_i < len(board) and 0 <= danger_j < len(board[0]) and board[danger_i][danger_j] != 1:
                            board[danger_i][danger_j] = 2

    return sum(k.count(0) for k in board)